### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by fncischen**
@ryanboris

### CODE REVIEW

**Comment by fncischen** [comment]
@ryanboris I now know why I was messing up with ComponentDidMount this afternoon (2/19/2019)!

- Because I did not use map() to create a new array to replace the old [] from this.state.data, but rather declare directly a PostContainer Component on the initial render() of the React Lifecycle, that PostContainer is forced to search for this.props.data.[attributename]. 

- Since those attributes have not been defined, because we did not define what this.props.data was (ComponentDidMount has not been called yet), React will crash.

- Thus, by using map(), on the initial render(), this.state.data can render ZERO PostContainer components, and will not be forced to search for this.props.data on the PostContainer. Then, after ComponentDidMount is called at the end of the React Life Cycle & this.state.data is declared, on the second frame, map() will have data from this.state.data to create an array of Posts that the PostContainer can create with.

Yay! I think this experience probably made me retain ComponentDidMount (React Lifecycle) much faster than if I did this correctly the first time.

**Comment by fncischen** [comment]
note to self -> will do refactoring to like event handler and adding comments tomorrow. but conceptually, things make sense now 👍

**Comment by fncischen** [comment]
notes: this topic will require extra review 👍

File: instagram/src/Components/CommentSection.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    componentDidMount() {
+        this.setState({
+            comments: this.props.comments
+        })
+    }
+
+    // addLike = e => {
+    //     if (this.state.liked == false) {
```
Comment (secantsquared) [inline instagram/src/Components/CommentSection.js]:
```suggestion
    //     if (!this.state.liked) //
    just a suggestion on how you can clean up the logic here.
```

File: instagram/src/Components/PostContainer.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import SearchBarHeader from "./Search";
+
+// import dummyData
+import dummyData from "./dummy-data";
+
+class PostContainer extends Component {
+
+    constructor(props) {
+        super(props);
+
```
Comment (secantsquared) [inline instagram/src/Components/PostContainer.js:23]: time saving suggestion - in modern and current React you can usually skip the constructor and super calls and just open with this...
```suggestion
state ={}
```

**Review by secantsquared — APPROVED** [review]
Looks like things are coming together! Great work!

**Comment by secantsquared** [comment]
@fncischen leaving the merging at your discretion due to the nature of the week long project

File: instagram/src/Components/PostContainer.js (hunk 2 above)
Comment (fncischen) [reply instagram/src/Components/PostContainer.js:23]: ooo thanks, i'll check it out!

**Comment by fncischen** [comment]
Yay, I got the Higher Order Components to work! @ryanboris 

To be honest, I'm not a fan of the arrow notation, and prefer the function(component A, component B, etc.), since it makes it a lot more easier to understand callback functions, like the tab components we did several weeks ago. I'm going to use that more often, since it makes far more sense to me.

**Comment by fncischen** [comment]
Next steps: styling.

**Pull request merged**